Files search interface (#4510)
mattermost/docs: source/help/getting-started/searching.rst

Searching
=========

> Use the Mattermost search box to find messages, replies, and file attachments across all channels you're a member of in your team's conversation history. File search is available now in Mattermost Cloud, and in Mattermost Self-Managed from v5.35, with mobile support coming soon.

Reviewer: ```suggestion
Use the Mattermost search box to find messages, replies, and file attachments across all channels you're a member of in your team's conversation history. File search is available now in Mattermost Cloud and in Mattermost Server from v5.35, with mobile support coming soon.
```

  
  .. image:: ../../images/channel-files-icon.png

All Searches
------------

- Multiple-word searches return results that contain *all* of your search criteria.
- Search modifiers can help narrow down searches. See the `Search Modifiers`_ section below for details.
- You can search Archived channels as long as you're a member of that channel. 

  - If you're unable to see messages in archived channels in your search results, ask your System Admin if **Allow users to view archived channels** has been disabled under **System Console > Site Configuration > Users and Teams**. 
  - To remove archived channels from your search results, you can leave Archived channels. 
- Like many search engines, common words such as ``the``, ``which``, and ``are`` (known as "stop words"), as well as two-letter and one-letter search terms, are not shown in search results because they typically return too many results. See the `Technical Notes`_ section below for details.
- IP addresses (e.g. ``10.100.200.101``) don't return results.
- When no search type is selected, message search results are returned first.

> .. Tip::

Reviewer: ```suggestion
.. tip::
```

  In the **Search Results** pane, you can switch between message and file search results by selecting the **Messages** and **Files** options.
  
  .. image:: ../../images/review-all-search-results.png

Searching for Messages 
----------------------

Select the search option, then select **Messages** as the search type, followed by your search criteria. 

.. image:: ../../images/search-messages.png

When message results display in the **Search Results** pane, select **Jump** to view the full message in context.

.. image:: ../../images/jump-to-message.png

Searching for Files
-------------------

> Select the search option, then select **Files** as the search type, followed by your search criteria. Search returns file attachments that match on file name or contain matching text content within supported document types. Each search result includes the file name, extension, size, as well as when and where the file was originally attached.

Reviewer: ```suggestion
Select the search option, then select **Files** as the search type, followed by your search criteria. The search returns file attachments that match on file name or contain matching text content within supported document types. Each search result includes the file name, extension, size, as well as when and where the file was originally attached.
```

.. image:: ../../images/search-files.png

> - For Mattermost Cloud workspaces, supported document file formats include PDF, DOCX, PPTX, ODT, HTML, and plain text documents. 

Author: @esethna @jespino  - Can Cloud end users search on RTF and PAGES file formats, as well as the contents of ZIP files as of April 21? What about file attachments shared prior to file search being available?

> - For Mattermost Self-Managed workspaces, supported document file formats include PDF, PPTX, ODT, HTML, and plain text documents. 

Reviewer: ```suggestion
- For Mattermost self-managed deployments, supported document file formats include PDF, PPTX, ODT, HTML, and plain text documents. 
```

>   - Your System Admin can `extend content search support to include DOCX, RTF, and PAGES file formats <https://docs.mattermost.com/administration/config-settings.html#enable-document-search-by-content>`__.

Reviewer: ```suggestion
  - System Admins can `extend content search support to include DOCX, RTF, and PAGES file formats <https://docs.mattermost.com/administration/config-settings.html#enable-document-search-by-content>`__.
```

>   - Your System Admin can also `extend content search support to include documents within ZIP files <https://docs.mattermost.com/administration/config-settings.html#enable-searching-content-of-documents-within-zip-files>`__. 

Reviewer: ```suggestion
  - System Admins can also `extend content search support to include documents within ZIP files <https://docs.mattermost.com/administration/config-settings.html#enable-searching-content-of-documents-within-zip-files>`__.
```

``before:``, ``after:``, and ``on:``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Use ``before:`` to find posts before a specified date.

Use ``after:`` to find posts after a specified date. 

Use both ``before:`` and ``after:`` together to search in a specified date range.  

Use ``on:`` to find posts from a specific date. Use the date picker to select a date, or type it in YYYY-MM-DD format. 

.. image:: ../../images/calendar2.png
  :width: 300 px

Examples:

-  Searching ``website before: 2018-09-01`` returns messages containing the term ``website`` posted prior to September 1, 2018.
-  Searching ``website after: 2018-08-01`` returns messages containing the term ``website`` posted after August 1, 2018.
-  Searching ``website before: 2018-09-01 after: 2018-08-01`` returns all messages containing the term ``website`` posted between August 1, 2018 and September 1, 2018.
-  Searching ``website on: 2018-09-01`` returns messages containing the term ``website`` posted on September 1, 2018.
  
Exclusions
^^^^^^^^^^

> Use the hyphen ``-`` symbol to exclude message terms from your search results. For example, searching ``test -release`` only returns results that include the term ``test`` that  do not contain the term ``release``.

Reviewer: ```suggestion
Use the hyphen ``-`` symbol to exclude message terms from your search results. For example, searching ``test -release`` only returns results that include the term ``test`` and exclude the term ``release``.
```

Valid hashtags:

- Don't start with a number.
- Are at least three characters long, excluding the ``#``.
- Are made up of alphanumeric or other unicode characters.
- May contain dots, dashes, or underscores.

To search for messages containing hashtags, select a hashtag in an existing post, or type the hashtag (including the pound ``#`` symbol) into the search bar. 

.. note::
  Hashtags don't link to channels. If you have a channel named “Marketing”, selecting a ``#marketing`` hashtag does not take you to the Marketing channel. To link to public channels, use the tilde ``~`` symbol followed by the channel name. For example ``~marketing``.

Technical Notes
---------------

Searching Chinese, Korean, and Japanese
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

- The best experience for searching in Chinese, Korean, and Japanese is to use MySQL 5.7.6 or later with special configuration. See the `Chinese, Japanese and Korean Search documentation <https://docs.mattermost.com/install/i18n.html>`__ for details.

> - You can perform searches without this configuration by adding wildcards ``*`` to the end of search terms.

Author: @esethna - this was previously worded as "You can search **to some degree** without this configuration..." (where emphasis is mine.)
Do we want to clarify what aspects of search will/won't work without the configuration?

Reviewer 2: Looks fine as you've written it
